modules/apps/forms-and-workflow/calendar/calendar-service/src/main/java/com/liferay/calendar/service/impl/CalendarServiceImpl.java

```diff
@@ -80,7 +80,7 @@ public String exportCalendar(long calendarId, String type)
 	public Calendar fetchCalendar(long calendarId) throws PortalException {
 		Calendar calendar = calendarPersistence.fetchByPrimaryKey(calendarId);
```

Comment:
Have you got exception from here ?

Reply:
Not exactly but close, in fact, the exception occurs at method check of CalendarPermission class (`CalendarPermission.check(getPermissionChecker(), calendar, ActionKeys.VIEW);`) (near line 87)
At this method, we have line
```
if (!contains(permissionChecker, calendar, actionId)) {
	throw new PrincipalException();
}
```
so, exeption is thrown when user does not have sufficient permission. To prevent exception from happening, we add checking permission condition for viewing new resource and return null when such condition is not satisfied in `fetchCalender` method (near line 83) of class `CalendarServiceImpl`

We have stacktrace of exception
```
Caused by: com.liferay.portal.kernel.security.auth.PrincipalException
	at com.liferay.calendar.service.permission.CalendarPermission.check(CalendarPermission.java:38)
	at com.liferay.calendar.service.impl.CalendarServiceImpl.fetchCalendar(CalendarServiceImpl.java:87)
```